beluga_tutorial/scripts/visualize.py

```diff
@@ -0,0 +1,162 @@
+
+    ax_landmark.set_title("Landmark Map")
+    ax_landmark.set_xlabel("States")
+    ax_landmark.set_xlim(-5, 100)
+    ax_landmark.set_ylim(0, 2)
+    ax_landmark.set_xticks(np.arange(-5, 101, 5))
+
+    ground_truth = yaml_data['simulation_records'][index]["ground_truth"]
+    ax_ground_truth = axs[5]
+    ax_ground_truth.clear()
+    ax_ground_truth.bar(ground_truth, 1, color='green')
+    ax_ground_truth.set_title(f"Ground Truth: {ground_truth}")
+    ax_ground_truth.set_xlabel("States")
+    ax_ground_truth.set_xlim(-5, 100)
+    ax_ground_truth.set_ylim(0, 2)
+    ax_ground_truth.set_xticks(np.arange(-5, 101, 5))
+
+    mean = "{:.3f}".format(yaml_data['simulation_records'][index]["estimation"]["mean"])
+    sd = "{:.3f}".format(yaml_data['simulation_records'][index]["estimation"]["sd"])
+    plt.text(
+        0.5,
+        0.5,
+        f"Mean: {mean}\nSD: {sd}",
+        ha='center',
+        va='center',
+        transform=axs[5].transAxes,
+        bbox=dict(facecolor='white', alpha=0.5),
+    )
+
+    plt.tight_layout()
+    plt.draw()
+
+
+def on_key(event, yaml_data, axs, num_frames):
+    if not hasattr(on_key, "current_frame"):
+        on_key.current_frame = 0
+
+    if event.key == 'right':
+        on_key.current_frame = (on_key.current_frame + 1) % num_frames
+    elif event.key == 'left':
+        on_key.current_frame = (on_key.current_frame - 1) % num_frames
+
+    plot_stages(yaml_data, axs, on_key.current_frame)
+
+
+def main(argv=None) -> int:
+    parser = argparse.ArgumentParser(description=globals()['__doc__'])
+
+    parser.add_argument(
+        '-p',
+        '--record-file-path',
+        type=str,
+        help='Absolute path to the record file generated by the beluga_tutorial example code',
+        required=True,
+    )
+
+    parser.add_argument(
+        '-m',
+        '--manual-control',
+        action='store_true',
+        help='Manual controlling the time steps of the visualization',
+    )
+
+    parser.add_argument(
+        '-i',
+        '--interval-ms',
+        type=int,
+        help='Delay between frames in milliseconds.',
+        required=False,
+        default=250,
+    )
+
+    args = parser.parse_args(argv)
+
+    with open(args.record_file_path, 'r') as file:
+        yaml_data = yaml.safe_load(file)
+
+    fig, axs = plt.subplots(6, 1)
+    num_frames = len(yaml_data['simulation_records'])
+
+    if args.manual_control:
+        plot_stages(yaml_data, axs, 0)
+        fig.canvas.mpl_connect(
+            'key_press_event',
+            lambda event: on_key(event, yaml_data, axs, num_frames),
+        )
+    else:
+
+        def update(current_frame):
+            plot_stages(yaml_data, axs, current_frame)
+
+        anim = animation.FuncAnimation(  # noqa: F841
+            fig,
+            update,
+            frames=num_frames,
+            blit=False,
+            interval=args.interval_ms,
+            repeat=False,
+        )
+
+    plt.show()
```

Comment:
I would suggest this to avoid messing with function object attributes manually, which is not good practice.
```python
    fig, axs = plt.subplots(6, 1)
    num_frames = len(yaml_data['simulation_records'])

    def plot_stages_update(current_frame: int) -> None:
        """Helper function to update plots."""
        plot_stages(yaml_data, axs, current_frame)

    if args.manual_control:
        current_frame = 0
        plot_stages_update(current_frame)

        def on_key(event):
            """Callback function to handle key events."""
            nonlocal current_frame

            if event.key == 'right':
                current_frame += 1
            elif event.key == 'left':
                current_frame -= 1

            current_frame %= num_frames
            plot_stages_update(current_frame)

        fig.canvas.mpl_connect('key_press_event', on_key)

    else:
        anim = animation.FuncAnimation(  # noqa: F841
            fig,
            plot_stages_update,
            frames=num_frames,
            blit=False,
            interval=args.interval_ms,
            repeat=False,
        )
```